moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp

```diff
@@ -90,167 +134,164 @@ void JogCalcs::startMainLoop(JogArmShared& shared_variables)
     position_filters_.emplace_back(parameters_.low_pass_filter_coeff);
   }
 
-  // Initialize the position filters to initial robot joints
-  while (!updateJoints(shared_variables) && ros::ok())
+  // initialization is finished
+  wait_for_jog_commands_ = true;
+  have_nonzero_twist_stamped_ = false;
+  have_nonzero_joint_jog_ = false;
+  have_nonzero_command_ = false;
+}
+
+void JogCalcs::start()
+{
+  stop_requested_ = false;
+  status_pub_ = nh_.advertise<std_msgs::Int8>(parameters_.status_topic, ROS_QUEUE_SIZE);
```

Comment:
I did this so you could wait on the status thread to be ready to signify everything was initialized.  I'm not sure this is good or standard.  Before it was waiting on the services but because I moved them into the constructor here it messed up the timing of some of the tests.  I think maybe the right approach is to wait for the first publish of status and put this back in the constructor.

Reply:
I was wrong, they already do wait for a first message and this doesn't need to be here.  I moved it with the other ros message interface stuff in the constructors.  Good catch.